tests/test_set.cpp

```diff
@@ -279,6 +279,22 @@ TEST_CASE("frozen::set of frozen::set", "[set]") {
   static_assert(ce.find(s1({0})) == ce.end(), "");
 }
 
+struct Foo {
+  int x;
+};
+constexpr inline bool operator==(Foo const &a, Foo const &b) { return a.x == b.x; }
+constexpr inline bool operator<(Foo const &a, Foo const &b) { return a.x < b.x; }
+constexpr inline bool operator<(Foo const &a, int b) { return a.x < b; }
+constexpr inline bool operator<(int a, Foo const &b) { return a < b.x; }
+
+TEST_CASE("frozen::set heterogeneous container", "[set]") {
+  constexpr auto set = frozen::make_set<Foo, std::less<>>({Foo{1}, Foo{2}, Foo{3}});
```

Comment:
could you also add a test that intializes from a plain C array? (for instance with the same content as `set` then we can add a check that tests for equality).

Reply:
Did I understand correctly that you mean a test that will check that `make_set` makes the same objects from a C array and `std::array`? Or just the same test as this, only with initialization from C array?